prebackup/prebackup.go

```diff
@@ -175,39 +201,42 @@ func (p *PreBackup) waitForReady(watcher watch.Interface) error {
 
 			last := p.getLastDeploymentCondition(deployment)
 
-			if last != nil {
-				// if the deadline can't be respected https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#progress-deadline-seconds
-				if last.Type == "Progressing" && last.Status == "False" && last.Reason == "ProgressDeadlineExceeded" {
-					watcher.Stop()
-					return fmt.Errorf("error starting pre backup pod %v: %v", deployment.GetName(), last.Message)
-				}
+			if last != nil && isDeadlineExceeded(last) {
+				return fmt.Errorf("error starting pre backup pod %v: %v", deployment.GetName(), last.Message)
 			}
 
-			// Wait until at least one replica is available and continue
-			if deployment.Status.AvailableReplicas > 0 {
+			if hasAvailableReplica(deployment) {
 				return nil
 			}
 
-			p.Log.Info("waiting for command pod to get ready", "name", deployment.GetName(), "namespace", deployment.GetNamespace())
+			p.Log.Info("waiting for command pod to get ready", "name", deployment.Name, "namespace", deployment.Namespace)
 
 		case watch.Error:
 
 			last := p.getLastDeploymentCondition(deployment)
 
 			if last != nil {
-				return fmt.Errorf("there was an error while starting pre backup pod %v: %v", deployment.GetName(), last.Message)
-
+				return fmt.Errorf("there was an error while starting pre backup pod '%v' in 'ns/%v': %v", deployment.Name, deployment.Namespace, last.Message)
```

Comment:
we should stick to common resource naming conventions (there are simillar other errors being thrown, I'm just mentioning here)
```suggestion
				return fmt.Errorf("there was an error while starting pre backup pod in deployment '%s/%s': %s", deployment.Namespace, deployment.Name, last.Message)
```

Reply:
I've changed this, because when I saw it in the operator I was like: _Cool, `k8up-test/pod-XXXX-xxxx`, I can just copy it to `kubectl logs k8up-test/pod-XXXX-xxxx`_. But no, that's not the way it works. So I'm not sure about the convention you mention either, because the main convention for a `/` in K8s in my opinion is `resourceKind/resourceName`, not `namespaceName/resourceName`.

How would you like me to proceed?

Reply:
I guess there are different kinds of convention, or no convention at all :) Most operators I saw so far use the `<namespace>/<resource>` notation to indicate the location of a resource. AFAIK the `<kind>/<resource>` is only used in the CLI. 
Neither approach is directly copy-pastable, as your current context might default to a different namespace than the one logged, so you'd need the `--namespace` flag too anyway.
So, choose your poison? I've just never seen the `ns/<namespace>` notation. and kubectl seems to agree with me:
```
$ k -n ns/default get pods
invalid namespace "ns/default": [may not contain '/']
```

Reply:
Like I said, I understood the `/` convention as `<kind>/<resource>`, so it'd be `kubectl get ns/default`, which works perfectly ;)

Nevertheless, I'll switch back to `<ns>/<resource>` if that's what other operators use.